client/src/main/java/org/eclipse/hono/client/device/amqp/TraceableCommandResponder.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.client.device.amqp;
+
+import java.util.Map;
+
+import org.eclipse.hono.client.ServiceInvocationException;
+
+import io.opentracing.SpanContext;
+import io.vertx.core.Future;
+import io.vertx.proton.ProtonDelivery;
+
+/**
+ * A sender to send back the response message of a command to Hono's AMQP adapter.
+ */
+public interface TraceableCommandResponder extends CommandResponder {
```

Comment:
I'm not quite sure what the question is aiming at. Hence the long answer. 
Why interfaces? 
The client provides an API that Hono users interact with directly and must therefore remain stable. In our efforts to make changes after the 1.0 release without breaking the API, I have found that this was very difficult, where we had no interfaces and worked directly with the implementations (not really surprising). So I made sure that the user only works with interfaces as much as possible.
Why is tracing not included in the interfaces?
First of all, I designed the interfaces for the AMQP adapter client without tracing, because it should be as simple and generally understandable as possible. In the typical use cases, no distributed tracing is needed because it only makes sense if the client and the Hono installation are operated by the same party. In general, I think that overloaded methods in APIs of libraries should be an exception.  However, Carsten asked me to provide tracing support anyway, because it can be helpful in some cases (e.g. for end-to-end testing of a Hono installation). I have hidden these a bit and would document them accordingly (I still have to do that. In Readme or developer guide).

Reply:
I see